fe/fe-core/src/main/java/org/apache/doris/catalog/Partition.java

```diff
@@ -500,4 +508,17 @@ public boolean convertRandomDistributionToHashDistribution(List<Column> baseSche
         }
         return hasChanged;
     }
+
+    private static void registerMetric(Partition partition) {
+        GaugeMetric<Long> gauge = new GaugeMetric<Long>("partition_query_count",
+                Metric.MetricUnit.NOUNIT, "partition query count") {
+            @Override
+            public Long getValue() {
+                return partition.partition_query_count.longValue();
+            }
+        };
+        gauge.addLabel(new MetricLabel("partition_name", partition.getName()));
+        gauge.addLabel(new MetricLabel("partition_id", String.valueOf(partition.getId())));
+        MetricRepo.addMetric(gauge);
```

Comment:
Are you sure to add gauge for each partition? There will be a huge number of metrics...

Reply:
Adding metrics for partition is to find zero-read partition for the consideration of cost. In a cluster of 30 DBS, every db has 30 tables, every table has 100 partitions. There will be 90000 partition metrics. That is acceptable. In BE, every tablet has 3 metrics. The number of tablet metric is much larger than partition metric.